source/uwp/Renderer/lib/AdaptiveElementParserRegistration.cpp

```diff
@@ -143,4 +147,13 @@ namespace AdaptiveNamespace
         std::shared_ptr<CustomElementWrapper> elementWrapper = std::make_shared<CustomElementWrapper>(cardElement.Get());
         return elementWrapper;
     }
+
+    HRESULT SharedModelElementParser::GetAdaptiveParserRegistration(_COM_Outptr_ IAdaptiveElementParserRegistration** elementParserRegistration)
+    {
+        ComPtr<IAdaptiveElementParserRegistration> parserRegistration;
+        RETURN_IF_FAILED(m_parserRegistration.As(&parserRegistration));
+        RETURN_IF_FAILED(parserRegistration.CopyTo(elementParserRegistration));
```

Comment:
not positive, so disregard if incorrect, but can't we just do:

```
RETURN_IF_FAILED(m_parserRegistration.CopyTo(IID_PPV_ARGS(elementParserRegistration)));
```

?

Reply:
Doesn't quite seem to work, as IID_PPV_ARGS gives you void** and we need IInspectable** here.

HRESULT Microsoft::WRL::WeakRef::CopyTo(IWeakReference **) noexcept const': cannot convert argument 2 from 'void **' to 'IInspectable **'

---
In reply to: [322489870](https://github.com/microsoft/AdaptiveCards/pull/3429#discussion_r322489870) [](ancestors = 322489870)

Reply:
Okay, fooling around with this a bit i got us down to:

return m_parserRegistration.CopyTo<IAdaptiveElementParserRegistration>(elementParserRegistration);

which seems better.

---
In reply to: [322496738](https://github.com/microsoft/AdaptiveCards/pull/3429#discussion_r322496738) [](ancestors = 322496738,322489870)